File: csharp/etl/Etl.cs
using System.Collections.Generic;

public static class Etl
{
    public static Dictionary<string, int> Transform(Dictionary<int, string[]> old)
    {
        var result = new Dictionary<string, int>();
        foreach (var (key, value) in old)
        {
            foreach (var t in value)
            {
                result.Add(t.ToLower(), key);
            }
        }

        return result;
    }
}

File: cpp/etl/etl.cpp
#include "etl.h"
#include <string>

namespace etl
{

    std::map<char, int> transform(std::map<int, std::vector<char>> old)
    {

        std::map<char, int> result{};

        for (auto [k, v] : old)
        {
            for (auto t : v)
            {

                t = static_cast<char>(std::tolower(static_cast<unsigned char>(t)));
                result.emplace(t, k);
            }
        }

        return result;
    }

} // namespace etl
